server/engine/engine.py

```diff
@@ -522,108 +569,127 @@ def preprocess_trials(self):
+                            trialSegmentName = f'{baseTrialName}_{iseg + 1}'
                         else:
-                            self.c3dFiles[baseTrialName] = baseC3DFile
+                            print(f' --> Trimmed time range: {segment[0]:1.2f} to {segment[1]:1.2f} s')
+                            trialSegmentName = baseTrialName
+
+                        # Create a new trial name for this segment.
+                        segmentTrialNames.append(trialSegmentName)
+
+                        # Create a new set of force plates for this segment.
+                        forcePlates = []
+                        for forcePlate in baseForcePlates:
+                            forcePlateCopy = nimble.biomechanics.ForcePlate.copyForcePlate(forcePlate)
+                            forcePlateCopy.trim(segment[0], segment[1])
+                            forcePlates.append(forcePlateCopy)
+                        segmentForcePlates.append(forcePlates)
+
+                        # Create a new set of timestamps for this segment.
+                        timestamps = []
+                        for timestamp in baseTimestamps:
+                            if segment[0] <= timestamp <= segment[1]:
+                                timestamps.append(timestamp)
+                        segmentTimestamps.append(timestamps)
+
+                        # Create a new set of frames per second for this segment.
+                        segmentFramesPerSecond.append(baseFramesPerSecond)
+
+                        # Create a new marker set for this segment.
+                        self.trialMarkerSet[trialSegmentName] = baseMarkerSet
+
+                        # Create a new marker trial for this segment.
+                        markerTrial = []
+                        for itime, timestamp in enumerate(baseTimestamps):
+                            if segment[0] <= timestamp <= segment[1]:
+                                markerTrial.append(baseMarkerTrial[itime])
+                        segmentMarkerTrials.append(markerTrial)
+
+                    # Insert the new segments into the list of trials.
+                    self.trialNames[itrial:itrial] = segmentTrialNames
+                    self.trialForcePlates[itrial:itrial] = segmentForcePlates
+                    self.trialTimestamps[itrial:itrial] = segmentTimestamps
+                    self.trialFramesPerSecond[itrial:itrial] = segmentFramesPerSecond
+                    self.markerTrials[itrial:itrial] = segmentMarkerTrials
+
+                    # If this trial is from a c3d file, then assign the original c3d file to the segments.
+                    if os.path.exists(c3dFilePath):
+                        baseC3DFile = self.c3dFiles.pop(trialName)
+                        for iseg, segment in enumerate(nonzeroForceSegments):
+                            if len(nonzeroForceSegments) > 1:
+                                self.c3dFiles[f'{baseTrialName}_{iseg + 1}'] = baseC3DFile
```

Comment:
Ok, this is not a right now problem, but this is a good reminder that we may want to eventually segment c3d files